activation/wire/challenge_v2.go

```diff
@@ -0,0 +1,42 @@
+package wire
+
+import (
+	"github.com/spacemeshos/go-spacemesh/codec"
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/hash"
+	"go.uber.org/zap/zapcore"
+)
+
+//go:generate scalegen
+
+// NIPostChallengeV2 collects information needed to generate a PoET challenge.
+// It's hash is the challenge registered in the PoET.
+type NIPostChallengeV2 struct {
+	PublishEpoch     types.EpochID
+	PrevATXID        types.ATXID
+	PositioningATXID types.ATXID
+	CommitmentATXID  *types.ATXID
```

Comment:
The commitment ATX is needed to be able to verify a PoST

in ATX v1 it used to be part of the NiPostChallenge, maybe so that PoET can refuse registrations with an invalid InitialPost (which would require the commitment ATX for verification). If we don't think that would be useful in the future we can move the commitment ATX to a different section of the ATX.

I generally would also prefer not having struct embedding on wire types. Instead the ATX could just be one big object with multiple fields and `NIPostChallengeV2` could be a method returning the struct above.

Reply:
`NIPostChallengeV2` is a structure aggregating things a miner must commit to before registering in the poet. The `InitialPost` already implicitly contains the commitment to a specific `commitment ATX` as it's one of the input parameters to PoST initialization so I think it doesn't need to be repeated (but it still needs to be present in the initial ATX to verify PoSTs and the VRF Nonce.

Not that the `NIPostChallengeV2` is not part of the ATX V2 at all.  During ATX handling, it can be reconstructed from the fields present in the ATX.